auth/src/main/java/com/firebase/ui/auth/ui/idp/SupportVectorDrawablesButton.java

```diff
@@ -0,0 +1,83 @@
+package com.firebase.ui.auth.ui.idp;
+
+import android.content.Context;
+import android.content.res.TypedArray;
+import android.graphics.drawable.Drawable;
+import android.os.Build;
+import android.support.v7.content.res.AppCompatResources;
+import android.support.v7.widget.AppCompatButton;
+import android.util.AttributeSet;
+
+import com.firebase.ui.auth.R;
+
+public class SupportVectorDrawablesButton extends AppCompatButton {
```

Comment:
This class seems to have some cool but non-obvious compatibility tricks.  Can you add comments (class level and in the one big method) explaining what's going on?